osu.Game.Rulesets.Catch/Edit/Blueprints/JuiceStreamSelectionBlueprint.cs

```diff
@@ -59,10 +87,38 @@ protected override void Update()
             pathCache.Validate();
         }
 
+        protected override void OnSelected()
+        {
+            initializeJuiceStreamPath();
+            base.OnSelected();
+        }
+
+        protected override bool OnMouseDown(MouseDownEvent e)
+        {
+            if (!IsSelected) return base.OnMouseDown(e);
+
+            switch (e.Button)
+            {
+                case MouseButton.Left when e.ControlPressed:
```

Comment:
`case...when` is a bit frowned upon in this repository, although I'm fine with it myself in simple cases.

Reply:
This is one of the few cases I don't have issue with this. Still unsure about usage because it's hard to make a call if/when it should be allowed or not.